flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -175,10 +176,10 @@ public Operation parseStatement(String sessionId, String statement)
         try {
             operations = context.wrapClassLoader(() -> parser.parse(statement));
         } catch (Exception e) {
-            throw new SqlExecutionException("Failed to parse statement: " + statement, e);
+            throw new SqlParseException("Failed to parse statement: " + statement, e);
```

Comment:
Personally, I think this pull request still not resolve the problem of the issue. You will still get the reason `org.apache.calcite.runtime.CalciteException: Non-query expression encountered in illegal context` when entering `invalid command;`. However, the reason message still can't help users figure out what happens. 

I think the purpose of this issue is reminding users this is an unsupported syntax. In order to improve the exception message, we can catch exception here, and wrap it into a new exception with message "Unsupported statement syntax is encountered" (or sth. similar) when the exception class is  `CalciteException` and the exception message is matched.